pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -265,15 +266,53 @@ func validatePipelineResults(results []PipelineResult) (errs *apis.FieldError) {
 		expressions = filter(expressions, looksLikeResultRef)
 		resultRefs := NewResultRefs(expressions)
 		if len(expressions) != len(resultRefs) {
-			errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("expected all of the expressions %v to be result expressions but only %v were", expressions, resultRefs),
+			return errs.Also(apis.ErrInvalidValue(fmt.Sprintf("expected all of the expressions %v to be result expressions but only %v were", expressions, resultRefs),
 				"value").ViaFieldIndex("results", idx))
 		}
 
+		if !TaskContainsResult(result.Value, taskMap) {
+			return errs.Also(apis.ErrInvalidValue("invalid propagation of finally task results to pipeline results",
+				"value").ViaFieldIndex("results", idx))
+		}
 	}
 
 	return errs
 }
 
+// put task names in a dictionary
+func convertTaskToTaskMap(tasks []PipelineTask) map[string]string {
```

Comment:
I don't think you need a map here, just a slice of task names.

Reply:
I am putting the slice of task names into a map for faster processing, otherwise I would have to loop through the task name every time I am trying to validate something.

Do you think this is the best approach, or should I go with the latter?

Reply:
how about sets.string? https://github.com/tektoncd/pipeline/blob/7b9eb0c553dedec05218c01ec8aa99378dee9361/vendor/k8s.io/apimachinery/pkg/util/sets/string.go#L30

Reply:
updated to use sets.